### PR TITLE
Add more commands

### PR DESCRIPTION
**Pull request by cryptoAtwill**
Add more rpc calls and cli commands for `fund`, `release`, `propagate` and `whitelist`.

### CODE REVIEW

File: src/cli/commands/manager/release.rs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#[derive(Debug, Args)]
+#[command(about = "Release operation in the gateway actor")]
+pub(crate) struct ReleaseArgs {
+    #[arg(long, short, help = "The JSON RPC server url for ipc agent")]
+    pub ipc_agent_url: Option<String>,
+    #[arg(long, short, help = "The address that leaves the subnet")]
+    pub from: Option<String>,
+    #[arg(long, short, help = "The subnet to release")]
+    pub subnet: String,
+    #[arg(help = "The amount to release")]
```
Comment (adlrocha) [inline src/cli/commands/manager/release.rs:52]:
```suggestion
    #[arg(help = "The amount to release in FIL")]
```

File: src/cli/commands/manager/fund.rs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+#[derive(Debug, Args)]
+#[command(about = "Fund operation in the gateway actor")]
+pub(crate) struct FundArgs {
+    #[arg(long, short, help = "The JSON RPC server url for ipc agent")]
+    pub ipc_agent_url: Option<String>,
+    #[arg(long, short, help = "The address that leaves the subnet")]
+    pub from: Option<String>,
+    #[arg(long, short, help = "The subnet to fund")]
+    pub subnet: String,
+    #[arg(help = "The amount to fund")]
```
Comment (adlrocha) [inline src/cli/commands/manager/fund.rs:52]:
```suggestion
    #[arg(help = "The amount to fund in FIL")]
```

File: src/server/handlers/manager/propagate.rs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+#[async_trait]
+impl JsonRPCRequestHandler for PropagateHandler {
+    type Request = PropagateParams;
+    type Response = ();
+
+    async fn handle(&self, request: Self::Request) -> anyhow::Result<Self::Response> {
+        let conn = match self.pool.get(&request.subnet) {
+            None => return Err(anyhow!("target parent subnet not found")),
```
Comment (adlrocha) [inline src/server/handlers/manager/propagate.rs:42]:
```suggestion
            None => return Err(anyhow!("target subnet not found")),
```

File: src/server/handlers/manager/release.rs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+#[async_trait]
+impl JsonRPCRequestHandler for ReleaseHandler {
+    type Request = ReleaseParams;
+    type Response = ();
+
+    async fn handle(&self, request: Self::Request) -> anyhow::Result<Self::Response> {
+        let conn = match self.pool.get(&request.subnet) {
+            None => return Err(anyhow!("target parent subnet not found")),
```
Comment (adlrocha) [inline src/server/handlers/manager/release.rs:42]:
```suggestion
            None => return Err(anyhow!("target subnet not found")),
```

File: src/server/handlers/manager/propagate.rs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            Some(conn) => conn,
+        };
+
+        let subnet = SubnetID::from_str(&request.subnet)?;
```
Comment (adlrocha) [inline src/server/handlers/manager/propagate.rs:46]: You came up with a convenient method in our previous PR, do you mind shifting these in all function to use that function? Thanks.

File: src/server/handlers/manager/fund.rs [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+            .parent()
+            .ok_or_else(|| anyhow!("no parent found"))?
+            .to_string();
+        let conn = match self.pool.get(&parent) {
+            None => return Err(anyhow!("target parent subnet not found")),
+            Some(conn) => conn,
+        };
+
+        let amount = TokenAmount::from_whole(request.amount);
+        let from = match request.from {
```
Comment (adlrocha) [inline src/server/handlers/manager/fund.rs:52]: Use function from previous PR

File: src/server/handlers/manager/whitelist.rs [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+            Some(conn) => conn,
+        };
+
+        let subnet = SubnetID::from_str(&request.subnet)?;
+        let to_add = request
+            .to_add
+            .iter()
+            .map(|s| Address::from_str(s))
+            .collect::<Result<Vec<_>, _>>()?;
+        let from = match request.from {
```
Comment (adlrocha) [inline src/server/handlers/manager/whitelist.rs:53]: Use function from previous PR

File: src/server/handlers/manager/whitelist.rs [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+#[async_trait]
+impl JsonRPCRequestHandler for WhitelistPropagatorHandler {
+    type Request = WhitelistPropagatorParams;
+    type Response = ();
+
+    async fn handle(&self, request: Self::Request) -> anyhow::Result<Self::Response> {
+        let conn = match self.pool.get(&request.subnet) {
+            None => return Err(anyhow!("target parent subnet not found")),
```
Comment (adlrocha) [inline src/server/handlers/manager/whitelist.rs:43]:
```suggestion
            None => return Err(anyhow!("target subnet not found")),
```

**Review by adlrocha — APPROVED** [review]
LGTM, thanks. Before you merge: 
- Be sure to apply the suggestions with fixes. 
- Use the convenient function to get the `from` address that you implemented in our previous PR. 
- Solve merge conflicts.

Thanks!

File: src/server/handlers/manager/propagate.rs (hunk 5 above)
Comment (cryptoAtwill) [reply src/server/handlers/manager/propagate.rs:46]: yep yep, this PR those functions are not in main yet, I will update this.

**Pull request merged**